lib/unleash/configuration.rb

```diff
@@ -23,7 +23,7 @@ class Configuration
       :bootstrap_config
 
     def initialize(opts = {})
-      ensure_valid_opts(opts)
+      validate_custom_http_headers!(opts[:custom_http_headers]) if opts.has_key?(:custom_http_headers)
```

Comment:
Total nit-picking: This is super duper valid, and good code. And I might even like it better than my own suggestion.

But a couple of lines below I use a different notation, so I think it would be a bit nice to be consistent with that. Feel free to ignore this feedback if you disagree.

```suggestion
      validate_custom_http_headers!(opts[:custom_http_headers]) unless opts[:custom_http_headers].nil?
```

Reply:
As someone who is just starting to learn Ruby the way it is now makes more sense at a quick glance, so I left it as it is for now. But I also agree that it should maybe change for consistency.